executor/builder.go

```diff
@@ -2164,6 +2167,102 @@ func (b *executorBuilder) buildIndexLookUpReader(v *plannercore.PhysicalIndexLoo
 	return ret
 }
 
+func buildNoRangeIndexMergeReader(b *executorBuilder, v *plannercore.PhysicalIndexMergeReader) (*IndexMergeReaderExecutor, error) {
+	partialPlanCount := len(v.PartialPlans)
+	partialReqs := make([]*tipb.DAGRequest, 0, partialPlanCount)
+	partialStreamings := make([]bool, 0, partialPlanCount)
+	indexes := make([]*model.IndexInfo, 0, partialPlanCount)
+	keepOrders := make([]bool, 0, partialPlanCount)
+	descs := make([]bool, 0, partialPlanCount)
+	feedbacks := make([]*statistics.QueryFeedback, 0, partialPlanCount)
+	ts := v.TablePlans[0].(*plannercore.PhysicalTableScan)
+	for i := 0; i < partialPlanCount; i++ {
+		feedback := statistics.NewQueryFeedback(0, nil, 0, ts.Desc)
+		feedback.Invalidate()
+		feedbacks = append(feedbacks, feedback)
+		tempReq, tempStreaming, err := b.constructDAGReq(v.PartialPlans[i])
+		if err != nil {
+			return nil, err
+		}
+		if is, ok := v.PartialPlans[i][0].(*plannercore.PhysicalIndexScan); ok {
+			keepOrders = append(keepOrders, is.KeepOrder)
+			descs = append(descs, is.Desc)
+			tempReq.OutputOffsets = []uint32{uint32(len(is.Index.Columns))}
+			collectIndex := false
+			tempReq.CollectRangeCounts = &collectIndex
+			partialReqs = append(partialReqs, tempReq)
+			partialStreamings = append(partialStreamings, tempStreaming)
+			indexes = append(indexes, is.Index)
+		} else {
+			ts := v.PartialPlans[i][0].(*plannercore.PhysicalTableScan)
+			keepOrders = append(keepOrders, ts.KeepOrder)
+			descs = append(descs, ts.Desc)
+			for i := range ts.Columns {
+				tempReq.OutputOffsets = append(tempReq.OutputOffsets, uint32(i))
+			}
+			collectTable := false
+			tempReq.CollectRangeCounts = &collectTable
+			partialReqs = append(partialReqs, tempReq)
+			partialStreamings = append(partialStreamings, tempStreaming)
+			indexes = append(indexes, nil)
+		}
+	}
+	tableReq, tableStreaming, err := b.constructDAGReq(v.TablePlans)
```

Comment:
This code block can be extracted as an individual function, so that buildNoRangeIndexLookUpReader can reuse it.

Reply:
it is from `2210` or  `for` block?

Reply:
2210~2221